src/main/kotlin/sipura/alg/GraphProperty.kt

```diff
@@ -114,4 +119,48 @@ object GraphProperty {
         }
         return true
     }
+
+    /**
+     * Calculates the diameter of the given graph [g] by performing a breadth-first-search from every vertex
+     * in the graph.
+     *
+     * Runtime: O(n * m)
+     * @return the diameter of the graph. 0 if the graph is empty. -1 if the graph is not connected.
+     */
+    fun <V> diameter(g: SimpleGraph<V>): Int {
+        if (g.V.isEmpty()) return 0
+        if (!Connectivity.isConnected(g)) return -1
+        var diameter = 0
+        for (v in g.V) {
+            var dist = -1
+            for (l in Traversal.breadthFirstSearchLayerIterator(g, v)) {
```

Comment:
initializing dist = -1 is a bit confusing IMO as I first thought this is for the case when the graph is disconnected, as described in the docs.
Do you think returning "diameter - 1" in the end makes it clearer, or do we keep the current version?

Reply:
I think the code is alredy pretty clear since the first sentence of the doumentation basically explains exactly what is being done. Also I think writing dist - 1 would be at least just as confusing.